api/context.go

```diff
@@ -142,7 +143,7 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if utils.GetSiteURL() == "" {
-		protocol := GetProtocol(r)
+		protocol := app.GetProtocol(r)
```

Comment:
Why is the GetProtocol in the app layer? isn't that related specifically to the api? I think its because you want to share this with apis v3 and v4?